Objects/TI4_Helpers/TI4_FactionHelper.ttslua

```diff
@@ -1122,6 +1126,703 @@ function tintTokens(params)
+
+        local dieType = "Die_10"
+        local removalDelay = 5
+        local radialOffset = 1.2
+        local heightOffset = 5
+        local dieSize = 1
+
+        --Finds a position, rotated around the Y axis, using distance you want + angle
+        --oPos is object pos, oRot=object rotation, distance = how far, angle = angle in degrees
+        local function _findGlobalPosWithLocalDirection(spawn_object, angle)
+            local object, distance = spawn_object, radialOffset * self.getScale().x
+            local oPos, oRot = object.getPosition(), object.getRotation()
+            local posX = oPos.x + math.sin( math.rad(angle+oRot.y) ) * distance
+            local posY = oPos.y + heightOffset
+            local posZ = oPos.z + math.cos( math.rad(angle+oRot.y) ) * distance
+            return {x=posX, y=posY, z=posZ}
+        end
+
+        --Gets a random rotation vector
+        local function _randomRotation()
+            --Credit for this function goes to Revinor (forums)
+            --Get 3 random numbers
+            local u1 = math.random();
+            local u2 = math.random();
+            local u3 = math.random();
+            --Convert them into quats to avoid gimbal lock
+            local u1sqrt = math.sqrt(u1);
+            local u1m1sqrt = math.sqrt(1-u1);
+            local qx = u1m1sqrt *math.sin(2*math.pi*u2);
+            local qy = u1m1sqrt *math.cos(2*math.pi*u2);
+            local qz = u1sqrt *math.sin(2*math.pi*u3);
+            local qw = u1sqrt *math.cos(2*math.pi*u3);
+            --Apply rotation
+            local ysqr = qy * qy;
+            local t0 = -2.0 * (ysqr + qz * qz) + 1.0;
+            local t1 = 2.0 * (qx * qy - qw * qz);
+            local t2 = -2.0 * (qx * qz + qw * qy);
+            local t3 = 2.0 * (qy * qz - qw * qx);
+            local t4 = -2.0 * (qx * qx + ysqr) + 1.0;
+            --Correct
+            if t2 > 1.0 then t2 = 1.0 end
+            if t2 < -1.0 then ts = -1.0 end
+            --Convert back to X/Y/Z
+            local xr = math.asin(t2);
+            local yr = math.atan2(t3, t4);
+            local zr = math.atan2(t1, t0);
+            --Return result
+            return {math.deg(xr),math.deg(yr),math.deg(zr)}
+        end
+
+        -- Use different color dice for manual auditability; copy from Multiroller
+        local unitTypeToDiceColor = {
+            ["Dreadnought"] = "Purple",
+            ["Flagship"] = "Black",
+            ["Destroyer"] = "Red",
+            ["War Sun"] = "Orange",
+            ["Carrier"] = "Blue",
+            ["Fighter"] = "Teal",
+            ["Infantry"] = "Green",
+            ["Cruiser"] = "Brown",
+            ["PDS"] = "Orange",
+            ["Space Dock"] = "Yellow",
+            ["Mech"] = "Pink"
+        }
+        
+        local diceToShip = {}
+        local unitTypesUsed = {}
+        _diceToBeRolled = {}
+
+        -- Spawn a die for each ship, around the system tile the ship is on.
```

Comment:
Dice can knock objects around.  Not sure about this on a busy table.

Reply:
I did make an effort to set the spawn radius (distance from the center of the system tile) such that dice spawn right along the edges. It's still possible to knock units around, but shouldn't change the tile they're on or do any significant rearranging. Even if a die lands so that's crooked because it's on a unit, getValue still seems to get a fair/accurate result.

Could try moving the dice off board, or just do a single system at a time, but I like that all the results from the ability are visible together.

Reply:
After seeing it on your demo table I'm not as worried, the dice seemed to drop straight down and not roll over the surface.  Maybe if they hit a unit they would, but seems ok for a first cut.